### PR TITLE
Update jobs docs

### LINKED ISSUE #600: docs and naming convention for new release process
It would be really helpful if docs could be updated with fuller descriptions on the release process and also name for stages of the release. 

There is reference [to it here](https://docs.opensafely.org/permissions/#releasing-outputs).

Naming the various stages e.g.
- L4
- Released outputs*
- Published outputs 
clearly i think is important as it has caused me a little confusion in semi-mixing up terms related to the old process and the new improved release

* I think this is a term in new process but just to mention, search of the docs also returns a result here https://docs.opensafely.org/repositories/#released_outputs

### PR DESCRIPTION
**Pull request by LFISHER7**
This extends the existing jobs site documentation to include information on the projects/workspaces/jobs hierarchy, the different types of outputs types (resolves #600) and viewing and publishing outputs.

There are a few details that I need to check are correct, so leaving in draft for now

### CODE REVIEW

**Comment by LisaHopcroft** [comment]
### General points

- Screenshots would be helpful for the user to orient themselves on the job site
- A diagram showing the relationship between project/workspace/job/action would also help

### Output types

- In **2. Released Outputs**: text to correct "These are analysis outputs are outputs"
- In **3. Draft public outputs**: change "close collaborators to your projects" to "close collaborators *of* your projects"
- In **3. Draft public outputs**: by "correct permissions" do you mean NHSE? If so, maybe amend the text to read "If you have the correct permissions" to "If you have approval from NHSE" just to be absolutely clear what we're meaning.

### Permissions

- Is it worth adding info on how to get these permissions if you need them?

### Creating a workspace

- Make "https://jobs.opensafely.org" a clickable URL
- `opensafely` -> `OpenSAFELY`
- `mintutes` -> `minutes`
- Add a full stop to this sentence: "When you add a new repository in the opensafely organisation, it may take up to 15 mintutes for it to be available to select at https://jobs.opensafely.org"

### Running your code on the server

- define "dependencies" (e.g. "If any of these actions have dependencies (i.e., ...) then they will also be run") or add a link to where this is defined elsewhere in the docs
- In this sentence "As such, jobs can sometimes take a while to start even if their dependencies have successfully completed" can we point the users to where they can see what is running on the server and the status page?

### Viewing analysis outputs on the server

- Make "https://jobs.opensafely.org/" a clickable URL

### Viewing released outputs

- I guess that some people reading this will not be copiloted, so the following text "please read [this section](https://www.opensafely.org/policies-for-researchers/#all-datasets-sharing) of the researcher policy document and contact your co-pilot" should be changed to read "please read [this section](https://www.opensafely.org/policies-for-researchers/#all-datasets-sharing) of the researcher policy document and/or contact your co-pilot *(if you have one)*" or something similar

### Publishing outputs

- Missing closing parenthesis in the first sentence
- Incorrectly formatted URL in the second sentence
- "You should also create draft public outputs for review" - this could be interpreted as asking users to make public drafts before they receive NHSE approval; maybe change this sentence to "Following approval from NHSE, you should create draft public outputs for review" or similar

File: docs/jobs-site.md
Comment (StevenMaude) [inline docs/jobs-site.md:11]: I haven't looked at this PR to review it specifically, but just to say that it's great people are using Mermaid where it's suitable over creating a diagram elsewhere :heart: :+1:

Comment (LFISHER7) [reply docs/jobs-site.md:11]: @LisaHopcroft introduce me to it, it's great!

**Comment by LisaHopcroft** [comment]
Looking great! The diagram and the screenshots really help.

When you introduce the concept of an action in the *Jobs site structure* section, is it worth saying that *actions* are defined in the `project.yaml` document? The text in the grey box around the actions could be amended to read "Actions (defined in project.yaml)" or something too?

Also, you could include an action that does not contribute towards the job to make it clear that subsets of actions can be run in one job - you don't have to run all actions in one job.

File: docs/jobs-site.md
Comment (bloodearnest) [inline docs/jobs-site.md:9]: So this is using `Job` and `Action` as the nouns to describe things in the system to users.

Internally, we call these a "Job Request" and a "Job". The "Action" is the specification in the project.yaml - a Job is an instance of an Action, basically.

My problem here is having 2 definitions of the noun "Job". I think it's already caused confusion.

I'm not sure sure what the solution is, but I'm just raising it for discussion.

Comment (LFISHER7) [reply docs/jobs-site.md:9]: Ah right, I've misconstrued that slightly. I'll try and reword this to be consistent with your ontology

Comment (LFISHER7) [reply docs/jobs-site.md:9]: Any objections to  surfacing the idea of a "Job request" in the docs?

Comment (bloodearnest) [reply docs/jobs-site.md:9]: Not from me, I think it it's a helpful term, as as the Request part gives more information.

**Comment by LFISHER7** [comment]
I've altered the text in the site structure section and have edited the flowchart which _hopefully_ clears things up

**Comment by LisaHopcroft** [comment]
Looking good!

I was going to suggest that it might be worth referring explicitly to the annotations in the screenshots (i.e., the red arrows/boxes). For example, `Click the Create a new workspace button (highlighted by the red box in the screenshot below)` or `Click the Run Jobs button from your workspace (highlighted by the red arrow in the screenshot below)`...what do you think?

**Review by LisaHopcroft — COMMENTED** [review]
The explicit mentioning of the red annotations would probably be most helpful in [this section](https://opensafely-documentation-pr-897.onrender.com/jobs-site/#viewing-and-requesting-changes-to-your-permissions). Not sure what the red arrow is pointing to there.

**Review by bloodearnest — APPROVED** [review]
This LGTM, and we really need something for some of these docs, so am approving and we can fix later if needed

**Pull request merged**
